atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java

```diff
@@ -252,21 +251,8 @@ static void checkAndSetReplicationFactor(Cassandra.Client client, KsDef ks, bool
         @Override
         public Boolean apply(Client client) throws UnsupportedOperationException {
             try {
-                String versionString = client.describe_version();
-                String[] components = versionString.split("\\.");
-                if (components.length != 3) {
-                    throw new UnsupportedOperationException(String.format("Illegal version of Thrift protocol detected; expected format '#.#.#', got '%s'", Arrays.toString(components)));
-                }
-                int majorVersion = Integer.parseInt(components[0]);
-                int minorVersion = Integer.parseInt(components[1]);
-                int patchVersion = Integer.parseInt(components[2]);
-
-                // see cassandra's 'interface/cassandra.thrift'
-                if (majorVersion >= 19 && minorVersion >= 37 && patchVersion >= 0) {
-                    return true;
-                } else {
-                    return false;
-                }
+                CassandraServerVersion serverVersion = new CassandraServerVersion(client.describe_version());
+                return serverVersion.supportsCheckAndSet();
```

Comment:
we should log this at INFO either here or in `com.palantir.atlasdb.keyvalue.cassandra.CassandraKeyValueService#init`

```
        boolean supportsCAS = clientPool.runWithRetry(CassandraVerifier.underlyingCassandraClusterSupportsCASOperations);
        log.info("Cassandra {} support CAS", (supportsCAS ? "does" : "does not"));
```